frontend/src/pages/UserProfile.js

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import { useState } from "react";
+import AppBar from '@mui/material/AppBar';
+import Button from '@mui/material/Button';
+import PersonIcon from '@mui/icons-material/Person';
+import CssBaseline from '@mui/material/CssBaseline';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import Typography from '@mui/material/Typography';
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+import { TextField } from "@mui/material";
+
+const DEFAULT_USERNAME = "USERNAME"
+const DEFAULT_PASSWORD = "PASSWORDa1!"
+const DEFAULT_FIRSTNAME = "BOB"
+const DEFAULT_LASTNAME = "TAN"
+const DEFAULT_EMAIL = "eg@example.com"
+
+// TODO remove, this demo shouldn't need to reset the theme.
+const defaultTheme = createTheme({
+  palette: {
+    mode: "dark",
+  },
+});
+
+export default function UserProfile() {
+  const [username, setUsername] = useState(DEFAULT_USERNAME);
```

Comment:
These fields are not changed as a user fills up the form. Can you use change handlers or some other mechanism to update the inputted fields? You can test whether the fields are updated by console.log in the handleSubmit